aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/analytics/AnalyticsClient.java

```diff
@@ -359,33 +359,33 @@ public void removeGlobalMetric(String eventType, String metricName) {
     }
 
     /**
-     * Adds the specified campaign attributes to events to track Campaign Analytic
+     * Adds the specified pinpoint attributes to events to track Campaign/Journey Analytic
      * <p>
      * You should not use this method as it will be called by the NotificationManager when the app is opened
      * from a push notification.
      *
-     * @param campaign the map with campaign attributes of the campaign received
+     * @param eventSourceAttributes the map with pinpointAttributes attributes of the pinpointAttributes received
      */
-    public void setCampaignAttributes(Map<String, String> campaign) {
-        if (campaign == null) {
-            log.warn("Null campaign attributes provided to setCampaignAttributes.");
+    public void setEventSourceAttributes(Map<String, String> eventSourceAttributes) {
```

Comment:
I'm not sold on the naming for this... "Event source attributes." Possibly, we could just skip the descriptors and get to just `setAttributes()`. or, `setPinpointAttributes()` ... `setGlobalAttributes()` .... are any of these better? Just throwing it out, I'm not sure. Let me know what you think.

Reply:
The thinking behind this was to keep it consistent with what's being done on the [iOS side](https://github.com/aws-amplify/aws-sdk-ios/pull/2387)

But maybe we don't need to do that??

Reply:
I'll take a look at the Pinpoint documentation to see if they use some sort of terminology that would be less verbose.

Reply:
I refactored some of the code a bit to make the logic a little clearer (maybe). I moved most of the code that deals with manipulating the attributes into the `AnalyticsClient`. The `NotificationClientBase` methods now just invokes `updateEventSourceGlobally` in `AnalyticsClient`